### PR TITLE
Allow health check timeout configuration

### PR DESCRIPTION
**Pull request by martin-tarjanyi**
On CI we sometimes see that the hardcoded 10 seconds is not enough for Hoverfly to become healthy. It would be nice to customize this value to stabilize our pipeline.

### CODE REVIEW

**Comment by martin-tarjanyi** [comment]
Sorry for the many changes because of the imports. I was not sure which is the expected layout.

**Review by tommysitu — APPROVED** [review]
Looking good! I'm wondering why hoverly takes longer than 10s to be ready in your case though.

File: src/main/java/io/specto/hoverfly/junit/core/Hoverfly.java [hunk 1]
```diff
@@ -533,16 +532,17 @@ private void persistSimulation(Path path, Simulation simulation) throws IOExcept
...
                 // TODO: prefer executors and tasks to threads
-                Thread.sleep(RETRY_BACKOFF_INTERVAL_MS);
+                Thread.sleep(hoverflyConfig.getHealthCheckRetryInterval().toMillis());
             } catch (InterruptedException e) {
                 throw new RuntimeException(e);
             }
         }
-        throw new IllegalStateException("Hoverfly has not become healthy in " + BOOT_TIMEOUT_SECONDS + " seconds");
+        throw new IllegalStateException(
+            "Hoverfly has not become healthy in " + hoverflyConfig.getHealthCheckTimeout().toMillis() + " milliseconds");
```
Comment (tommysitu) [inline src/main/java/io/specto/hoverfly/junit/core/Hoverfly.java:545]: prefer to print seconds here, it's more user friendly.

**Comment by martin-tarjanyi** [comment]
> Looking good! I'm wondering why hoverly takes longer than 10s to be ready in your case though.

Thanks! I'm not sure either, it happens very rarely

**Pull request merged**